packages/report-server/src/reportBuilder/functions/basic.ts

```diff
@@ -21,22 +21,22 @@ export const last = (values: FieldValue[]): FieldValue => {
   return values[values.length - 1];
 };
 
-export const eq = (value1, value2): boolean => {
-  return value1 === value2;
+export const eq = (val1: any, val2: any): boolean => {
+  return val1 === val2;
 };
 
-export const notEq = (value1, value2): boolean => {
-  return value1 !== value2;
+export const notEq = (val1: any, val2: any): boolean => {
+  return val1 !== val2;
 };
 
-export const gt = (value1, value2): boolean => {
-  return value1 > value2;
+export const gt = (val1: any, val2: any): boolean => {
+  return val1 > val2;
 };
 
-export const exists = (value: FieldValue): boolean => {
-  return value !== undefined;
+export const exists = (val: any): boolean => {
+  return val !== undefined;
 };
 
-export const notExists = (value: FieldValue): boolean => {
-  return value === undefined;
+export const notExists = (val: FieldValue): boolean => {
+  return val === undefined;
```

Comment:
Sorry I didn't realise we have basic functions at the top of everything here and I added something similar [here...](https://github.com/beyondessential/tupaia/blob/72ea0a8a8e4d3ba2f033e57c1806e23f19f131fd/packages/report-server/src/reportBuilder/transform/functions/aggregate/aggregations.ts#L8). @rohan-bes Do you think we can merge them together?

Reply:
So that function there is actually a function we have available when running expressions, eg.
```
{
  "transform": "select",
  "'Has No Data'": "notExists($row.dataValue)"
}
```

It shouldn't really be used in-code the way that your `isNotUndefined` function is being used 👍 

However, you do make a good point that:
a) It is hard to know which functions are available when parsing expressions
b) We often define the same utility functions over and over in typescript, eg. something like `isNotUndefined` exists in a couple of places

Regarding a) it's a bit of a tricky issue, ideally we'd move all the functions into `@tupaia/expression-parser` and then there would be one central place to look them all up! Even better, when we get a UX for creating reports via the admin panel we can auto-suggestion!

Regarding b) I think a `@tupaia/ts-utils` package is probably overdue at this point... We just haven't got around to creating it as of yet...